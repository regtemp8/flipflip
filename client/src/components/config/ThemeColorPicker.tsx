import * as React from 'react'

import { Fab, Grid, TextField, type Theme } from '@mui/material'

import createStyles from '@mui/styles/createStyles'
import withStyles, { type WithStyles } from '@mui/styles/withStyles'

import * as color from '@mui/material/colors'
import type ReduxProps from '../common/ReduxProps'
import { useAppDispatch, useAppSelector } from '../../store/hooks'

const styles = (theme: Theme) =>
  createStyles({
    colorGrid: {
      width: 170
    },
    colorButton: {
      backgroundColor: theme.palette.common.white,
      marginTop: 0,
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1),
      marginBottom: theme.spacing(1),
      boxShadow: 'none'
    },
    colorPickerButton: {
      backgroundColor: theme.palette.common.white,
      marginRight: theme.spacing(0.25),
      width: theme.spacing(2),
      height: theme.spacing(2),
      minHeight: theme.spacing(2),
      boxShadow: 'none'
    },
    colorField: {
      width: 100
    }
  })

const colors = [
  color.red,
  color.pink,
  color.purple,
  color.deepPurple,
  color.indigo,
  color.blue,
  color.lightBlue,
  color.cyan,
  color.teal,
  color.green,
  color.lightGreen,
  color.lime,
  color.yellow,
  color.amber,
  color.orange,
  color.deepOrange,
  color.brown,
  color.grey,
  color.blueGrey
]

export interface ThemeColorPickerProps
  extends ReduxProps<string>,
    WithStyles<typeof styles> {}

function ThemeColorPicker(props: ThemeColorPickerProps) {
  const dispatch = useAppDispatch()
  const currentColor = useAppSelector(props.selector)

  const onChangeBlack = () => {
    const black = color.common.black
    const grey = color.grey
    setColorChange({
      50: grey[50],
      100: grey[100],
      200: grey[200],
      300: grey[300],
      400: grey[400],
      500: grey[500],
      600: grey[600],
      700: grey[700],
      800: grey[800],
      900: grey[900],
      A100: grey.A100,
      A200: grey.A200,
      A400: grey.A400,
      A700: grey.A700,
      main: black
    })
  }

  const onChangeWhite = () => {
    const white = color.common.white
    const grey = color.grey
    setColorChange({
      50: grey[900],
      100: grey[800],
      200: grey[700],
      300: grey[600],
      400: grey[500],
      500: grey[400],
      600: grey[300],
      700: grey[200],
      800: grey[100],
      900: grey[50],
      A100: grey.A700,
      A200: grey.A400,
      A400: grey.A200,
      A700: grey.A100,
      main: white
    })
  }

  const onChangeColor = (color: any) => {
    setColorChange({
      50: color[50],
      100: color[100],
      200: color[200],
      300: color[300],
      400: color[400],
      500: color[500],
      600: color[600],
      700: color[700],
      800: color[800],
      900: color[900],
      A100: color.A100,
      A200: color.A200,
      A400: color.A400,
      A700: color.A700,
      main: color[500]
    })
  }

  const setColorChange = (color: any) => {
    dispatch(props.action(color))
  }

  const classes = props.classes
  return (
    <Grid container alignItems="center">
      <Grid item className={classes.colorGrid}>
        <Fab
          className={classes.colorButton}
          style={{ backgroundColor: currentColor }}
          size="medium"
        >
          <div />
        </Fab>
        <TextField
          variant="standard"
          className={classes.colorField}
          label="Color"
          InputProps={{
            readOnly: true
          }}
          value={currentColor}
        />
      </Grid>
      <Grid item xs={12} sm>
        <Grid container alignItems="center">
          {colors.map((c) => (
            <Grid key={c[500]} item>
              <Fab
                className={classes.colorPickerButton}
                style={{ backgroundColor: c[500] }}
                value={c[500]}
                onClick={() => onChangeColor(c)}
                size="small"
              >
                <div />
              </Fab>
            </Grid>
          ))}
          <Grid key={color.common.white} item>
            <Fab
              className={classes.colorPickerButton}
              style={{ backgroundColor: color.common.white }}
              value={color.common.white}
              onClick={onChangeWhite}
              size="small"
            >
              <div />
            </Fab>
          </Grid>
          <Grid key={color.common.black} item>
            <Fab
              className={classes.colorPickerButton}
              style={{ backgroundColor: color.common.black }}
              value={color.common.black}
              onClick={onChangeBlack}
              size="small"
            >
              <div />
            </Fab>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}

;(ThemeColorPicker as any).displayName = 'ThemeColorPicker'
export default withStyles(styles)(ThemeColorPicker as any)
