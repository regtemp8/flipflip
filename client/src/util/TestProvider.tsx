import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
