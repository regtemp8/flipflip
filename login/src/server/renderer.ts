import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
