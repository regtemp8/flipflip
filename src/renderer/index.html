<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
