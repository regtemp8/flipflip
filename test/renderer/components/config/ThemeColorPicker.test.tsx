import React from "react";
import { describe, it, expect } from "@jest/globals";
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import renderer from "react-test-renderer";
import ThemeColorPicker from "../../../../src/renderer/components/config/ThemeColorPicker";
import TestProvider from "../../../util/TestProvider";

describe("ThemeColorPicker", () => {
  it("should render without color when currentColor is null", () => {
    const component = renderer.create(
      <TestProvider>
        <ThemeColorPicker currentColor={null} onChangeColor={(e: any) => {}} />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render color when currentColor has value", () => {
    const component = renderer.create(
      <TestProvider>
        <ThemeColorPicker currentColor="#f00" onChangeColor={(e: any) => {}} />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should change color and input when a color preset button is clicked", async () => {
    const onChange = jest.fn()
    const { container } = render(
      <TestProvider>
        <ThemeColorPicker currentColor="#f00" onChangeColor={onChange} />
      </TestProvider>
    );
    
    const color = '#9c27b0'
    const button = container.querySelector('button[value="' + color + '"]')
    fireEvent.click(button)
    
    expect(onChange.mock.calls).toHaveLength(1);
    expect(onChange.mock.calls[0][0].main).toBe(color);

  });
  it("should not allow input changes", async () => {
    render(
      <TestProvider>
        <ThemeColorPicker currentColor="#f00" onChangeColor={(e: any) => {}} />
      </TestProvider>
    );

    const input = screen.getByRole('textbox')
    expect(input.hasAttribute('readonly')).toBe(true)
  });
  it("should not show color picker when color is clicked", async () => {
    const {container} = render(
      <TestProvider>
        <ThemeColorPicker currentColor='#f00' onChangeColor={(e: any) => {}} />
      </TestProvider>
    );

    const button = container.querySelector('button[class*=" ThemeColorPicker-colorButton-"]') as HTMLButtonElement
    expect(button.onclick).toBe(null)
  });
});
