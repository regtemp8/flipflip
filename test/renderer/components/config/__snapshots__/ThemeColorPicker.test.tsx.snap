// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeColorPicker should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item ThemeColorPicker-colorGrid-1 css-13i4rnv-MuiGrid-root"
  >
    <button
      className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeMedium MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeMedium MuiFab-default ThemeColorPicker-colorButton-2 css-1fwudth-MuiButtonBase-root-MuiFab-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "backgroundColor": "#fff",
        }
      }
      tabIndex={0}
      type="button"
    >
      <div />
    </button>
    <div
      className="MuiFormControl-root MuiTextField-root ThemeColorPicker-colorField-4 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-syr9i-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
      >
        Color
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-cwspju-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input Mui-readOnly MuiInputBase-readOnly css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={true}
          required={false}
          type="text"
          value="#fff"
        />
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true css-cyocdy-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#f44336",
            }
          }
          tabIndex={0}
          type="button"
          value="#f44336"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#e91e63",
            }
          }
          tabIndex={0}
          type="button"
          value="#e91e63"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#9c27b0",
            }
          }
          tabIndex={0}
          type="button"
          value="#9c27b0"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#673ab7",
            }
          }
          tabIndex={0}
          type="button"
          value="#673ab7"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#3f51b5",
            }
          }
          tabIndex={0}
          type="button"
          value="#3f51b5"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#2196f3",
            }
          }
          tabIndex={0}
          type="button"
          value="#2196f3"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#03a9f4",
            }
          }
          tabIndex={0}
          type="button"
          value="#03a9f4"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#00bcd4",
            }
          }
          tabIndex={0}
          type="button"
          value="#00bcd4"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#009688",
            }
          }
          tabIndex={0}
          type="button"
          value="#009688"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#4caf50",
            }
          }
          tabIndex={0}
          type="button"
          value="#4caf50"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#8bc34a",
            }
          }
          tabIndex={0}
          type="button"
          value="#8bc34a"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#cddc39",
            }
          }
          tabIndex={0}
          type="button"
          value="#cddc39"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#ffeb3b",
            }
          }
          tabIndex={0}
          type="button"
          value="#ffeb3b"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#ffc107",
            }
          }
          tabIndex={0}
          type="button"
          value="#ffc107"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#ff9800",
            }
          }
          tabIndex={0}
          type="button"
          value="#ff9800"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#ff5722",
            }
          }
          tabIndex={0}
          type="button"
          value="#ff5722"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#795548",
            }
          }
          tabIndex={0}
          type="button"
          value="#795548"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#9e9e9e",
            }
          }
          tabIndex={0}
          type="button"
          value="#9e9e9e"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#607d8b",
            }
          }
          tabIndex={0}
          type="button"
          value="#607d8b"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#fff",
            }
          }
          tabIndex={0}
          type="button"
          value="#fff"
        >
          <div />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ThemeColorPicker-colorPickerButton-3 css-12m9wit-MuiButtonBase-root-MuiFab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "#000",
            }
          }
          tabIndex={0}
          type="button"
          value="#000"
        >
          <div />
        </button>
      </div>
    </div>
  </div>
</div>
`;
